examples/set-label/functions/fn-config.yaml

```diff
@@ -8,5 +8,12 @@ metadata:
         image: gcr.io/kpt-functions/set-label
     config.kubernetes.io/local-config: 'true'
 data:
-  label_name: color
-  label_value: orange
+  specs:
```

Comment:
I saw you use `rn.Pipe(kyaml.Lookup("data", "specs"))` below.
Why don't we define a corresponding go struct and simply use unmarshal? IMO the latter approach is easier to reasonable about.

Reply:
I don't think we should implement our own struct for function config given this part may be used by other functions and they have already had well-defined utilities in kyaml.

Reply:
Sorry for not being very clear.
I meant `data` is standard field for `ConfigMap`, but `data.specs` is not. `kyaml.Lookup("data", "specs")` this feels like hardcoding the path.
IMO it's better to treat whatever under `.data` as a blob and unmarshal the whole blob to `addLabelSpecs`.

`addLabelSpecs` may look like this:
```
type addLabelSpecs struct {
	Labels []addLabelSpec `json:"labels,omitempty" yaml:"labels,omitempty"`
	FieldSpec types.FieldSpec `json:"fieldSpec,omitempty" yaml:"fieldSpec,omitempty"` // Field config for all labels
}

type addLabelSpec struct { // you already defined this in functions/go/set-label/main.go
	LabelName  string            `json:"label_name,omitempty" yaml:"label_name,omitempty"`
	LabelValue string            `json:"label_value,omitempty" yaml:"label_value,omitempty"`
	FieldSpecs []types.FieldSpec `json:"fieldSpecs,omitempty" yaml:"fieldSpecs,omitempty"` // Field config for one label.
}
```
Then you have something like:
```
d := kyaml.Lookup("data")
yaml.Unmarshal(d, &ads)
```

Does it make sense to you?

Reply:
done